includes/Handlers/Checkout.php

```diff
@@ -0,0 +1,146 @@
+                'permission_callback' => '__return_true',
+            )
+        );
+	}
+
+	/**
+     * Redirects to the checkout page.
+     * @since 3.3.2
+     */
+    public function redirect_to_checkout() {
+		$this->add_multiple_items_and_apply_coupon();
+        exit;
+    }
+
+    private function add_multiple_items_and_apply_coupon() {
+        $product_ids = isset($_REQUEST['products']) ? array_map('trim', explode(',', urldecode(wp_unslash($_REQUEST['products'])))) : array();
+        $quantities = isset($_REQUEST['quantity']) ? array_map('trim', explode(',', urldecode(wp_unslash($_REQUEST['quantity'])))) : array();
+        $coupon_code = isset($_REQUEST['coupon']) ? wp_unslash($_REQUEST['coupon']) : '';
+        $clear_cart = isset($_REQUEST['clear']) ? wp_unslash($_REQUEST['clear']) : false;
+
+        // Clear the existing WooCommerce cart
+        if($clear_cart !== false)
+        {
+            WC()->cart->empty_cart();
+        }
+
+
+        $was_added_to_cart = false;
+        $url = false; // Ensure $url is defined
+
+        // Ensure we have the same number of quantities as product IDs
+        $quantities = array_pad($quantities, count($product_ids), 1);
+
+        foreach ($product_ids as $index => $product_param) {
+            $product_data = explode( ':', $product_param );
+
+            if ( count( $product_data ) != 2 ) {
+                continue;
+            }
+            $product_sku = sanitize_text_field( $product_data[0] );
+            $quantity = intval( $product_data[1] );
+            if ( $quantity <= 0 ) {
+                continue;
+            }
+            // Get the product ID by SKU
+            // Extract only the last numeric part after the last underscore
+            $product_id = $this->get_product_id($product_sku);
+            $product_id = apply_filters('woocommerce_add_to_cart_product_id', absint($product_id));
+            $adding_to_cart = wc_get_product($product_id);
+
+            if (!$adding_to_cart) {
+                continue;
+            }
+
+            $add_to_cart_handler = apply_filters('woocommerce_add_to_cart_handler', $adding_to_cart->get_type(), $adding_to_cart);
+
+            if ('grouped' === $add_to_cart_handler)
+            {
+                continue;
+            }
+
+            $passed_validation = apply_filters('woocommerce_add_to_cart_validation', true, $product_id, $quantity);
+                                wc_load_cart();
+                                $cart = WC()->cart;
+
+            if ($passed_validation && false !== $cart->add_to_cart($product_id, $quantity)) {
+                wc_add_to_cart_message(array($product_id => $quantity), true);
+                $was_added_to_cart = true;
+            } else {
+                $was_added_to_cart = false;
+            }
+        }
+
+        if ($coupon_code && !$cart->has_discount($coupon_code)) {
+            $cart->add_discount(trim($coupon_code));
+        }
```

Comment:
how much extra effort would it be to support a list of coupon codes rather than a single one? Initially we'd only be sending 1 but it would be good to futureproof it since Woo does support multiple